### PR TITLE
Go: Embedding Interfaces

### PR DESCRIPTION
**Pull request by rusher2004**
Adding my first lil bit of writing. It's a blog about embedding interfaces in Go, and enabling good, easy tests with it.

It comes with new unicorn data too, of course.

### CODE REVIEW

File: content/blog/go-embedding-interfaces/index.md [hunk 1]
```diff
@@ -0,0 +1,373 @@
+---
+{
+    title: 'Embedding Interfaces in Go',
+    published: '2024-03-04',
+    authors: ['rusher2004'],
+    tags: ['go', 'testing'],
+    description: ""
```
Comment (crutchcorn) [inline content/blog/go-embedding-interfaces/index.md:7]: Hey @rusher2004 - let's add a description here (no more than 160 chars please!) and then we can ship :)

File: content/blog/go-embedding-interfaces/index.md [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+
+#### error
+
+![](./error_interface_def.png)
+
+#### fmt.Stringer
+
+![](./stringer_interface_def.png)
+
+## CHANGE THIS HEADING
```
Comment (crutchcorn) [inline content/blog/go-embedding-interfaces/index.md:67]: We probably wanna change this heading before going live too ;)

**Review by crutchcorn — CHANGES_REQUESTED** [review]

File: content/blog/go-embedding-interfaces/index.md (hunk 2 above)
Comment (rusher2004) [reply content/blog/go-embedding-interfaces/index.md:67]: Ha whoops, I could have sworn I had.

File: content/blog/go-embedding-interfaces/index.md (hunk 1 above)
Comment (rusher2004) [reply content/blog/go-embedding-interfaces/index.md]: 159 characters 😎

File: content/blog/go-embedding-interfaces/index.md [hunk 3]
```diff
@@ -0,0 +1,385 @@
+---
+{
+    title: 'Embedding Interfaces in Go',
+    published: '2024-03-04',
```
Comment (crutchcorn) [inline content/blog/go-embedding-interfaces/index.md:4]:
```suggestion
    published: '2024-03-06',
```

**Pull request merged**